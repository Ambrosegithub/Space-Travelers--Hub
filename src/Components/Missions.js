/* eslint-disable */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Container from 'react-bootstrap/Container';
import Table from 'react-bootstrap/Table';
import Button from 'react-bootstrap/Button';
import { fetchMissions, joinMission, leaveMission } from '../redux/missionFetch';

const Missions = () => {
    // get missions data from the store
    const dispatch = useDispatch();
    const missions = useSelector((state) => state.missions);
  
    useEffect(() => {
      if (!missions.length) {
        dispatch(fetchMissions);
      }
    },[]);
const handleJoinMission = (id) => dispatch(joinMission(id));
const handleLeaveMission = (id) => dispatch(leaveMission(id));
  return (
    <Container className="container-fluid">
    <Table className="col-xl-12 col-lg-12 col-md-12 col-sm-12">
        <thead>
          <tr>
            <th>Mission</th>
            <th>description</th>
            <th>Status</th>
            <th>{" "}</th>
          </tr>
        </thead>
        <tbody >
          {missions.map(({
            id, name, description, reserved
          }) => (
            <tr className='wrapper' key={id}>
            <td>
              <h3>{name}</h3>
            </td>
            <td>
              <p>{description}</p>
            </td>
            <td className="badge-cont">
                {reserved && <span className="act-badge">Active Member</span>}
                {!reserved && <span className="badge" style={{ color: "green" }}>NOT A MEMBER</span>}
              </td>
              <td className="button-cont">
                {reserved && (
                <Button 
                  className="leave-btn"
                  size="sm" 
                  variant="outline-danger"
                  onClick={() => handleLeaveMission(id)}>
                  Leave Mission
                </Button>
                )}
                {!reserved && (
                <Button 
                  className="join-btn"
                  size="sm" 
                  variant="outline-secondary"
                  onClick={() => handleJoinMission(id)}>
                  Join Mission
                </Button>
                )}
              </td>
          </tr>
          ))}
        </tbody>
      </Table>
    </Container>
  )
};

export default Missions;
