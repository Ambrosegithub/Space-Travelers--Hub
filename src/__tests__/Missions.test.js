import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Missions from '../Components/Missions';
import store from '../redux/configureStore';

describe('Test of Profile Components', () => {
  it('Profile Components Renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
