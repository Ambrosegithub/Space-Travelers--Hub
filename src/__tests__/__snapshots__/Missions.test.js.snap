// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Profile Components Profile Components Renders correctly 1`] = `
<div
  className="container-fluid container"
>
  <table
    className="col-xl-12 col-lg-12 col-md-12 col-sm-12 table"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
